HLTriggerOffline/Btag/src/HLTBTagPerformanceAnalyzer.cc

```diff
@@ -168,122 +168,127 @@ void HLTBTagPerformanceAnalyzer::dqmBeginRun(const edm::Run& iRun, const edm::Ev
-	if (TriggerResulsHandler.isValid())   trigRes=true;
-	if ( !trigRes ) { edm::LogInfo("NoTriggerResults") << "TriggerResults ==> not readable"; return;}
-	const TriggerResults & triggerResults = *(TriggerResulsHandler.product());
+  iEvent.getByToken(hlTriggerResults_, TriggerResulsHandler);
+  if (TriggerResulsHandler.isValid())   trigRes=true;
+  if ( !trigRes ) { edm::LogInfo("NoTriggerResults") << "TriggerResults ==> not readable"; return;}
+  const TriggerResults & triggerResults = *(TriggerResulsHandler.product());
 	
-	//get partons
-	if (m_mcMatching && !m_mcPartons_Label.empty() && m_mcPartons_Label != "NULL" ) {
-		iEvent.getByToken(m_mcPartons, h_mcPartons);
-		if (h_mcPartons.isValid()) MCOK=true;
-	}
+  //get partons
+  if (m_mcMatching && !m_mcPartons_Label.empty() && m_mcPartons_Label != "NULL" ) {
+    iEvent.getByToken(m_mcPartons, h_mcPartons);
+    if (h_mcPartons.isValid()) MCOK=true;
+  }
 
-	//fill the 1D and 2D DQM plot
-	Handle<reco::JetTagCollection> JetTagHandler;
-	for (unsigned int ind=0; ind<hltPathNames_.size();ind++) {
-		bool BtagOK=false;
-		JetTagMap JetTag;
-		if ( !_isfoundHLTs[ind]) continue;	//if the hltPath is not in the event, skip the event
-		if ( !triggerResults.accept(hltPathIndexs_[ind]) ) continue;	//if the hltPath was not accepted skip the event
+  //fill the 1D and 2D DQM plot
+  Handle<reco::JetTagCollection> JetTagHandler;
+  for (unsigned int ind=0; ind<hltPathNames_.size();ind++) {
+    bool BtagOK=false;
+    JetTagMap JetTag;
+    if ( !_isfoundHLTs[ind]) continue;	//if the hltPath is not in the event, skip the event
+    if ( !triggerResults.accept(hltPathIndexs_[ind]) ) continue;	//if the hltPath was not accepted skip the event
 		
-		//get JetTagCollection
-		if (!JetTagCollection_Label[ind].empty() && JetTagCollection_Label[ind] != "NULL" )
-		{
-			iEvent.getByToken(JetTagCollection_[ind], JetTagHandler);
-                        iEvent.getByToken(shallowTagInfosTokenPf_, shallowTagInfosPf);
-                        iEvent.getByToken(shallowTagInfosTokenCalo_, shallowTagInfosCalo);
-			if (JetTagHandler.isValid())   BtagOK=true;
-		}
+    //get JetTagCollection
+    if (!JetTagCollection_Label[ind].empty() && JetTagCollection_Label[ind] != "NULL" )
+      {
+        iEvent.getByToken(JetTagCollection_[ind], JetTagHandler);
+        iEvent.getByToken(shallowTagInfosTokenPf_, shallowTagInfosPf);
+        //                        iEvent.getByToken(shallowTagInfosTokenCalo_, shallowTagInfosCalo);
+        if (JetTagHandler.isValid() )   BtagOK=true;
+      }
 
-		//fill JetTag map
-		if (BtagOK) for ( auto  iter = JetTagHandler->begin(); iter != JetTagHandler->end(); iter++ )
-		{
-			JetTag.insert(JetTagMap::value_type(iter->first, iter->second));
-		}
-		else {
-		    edm::LogInfo("NoCollection") << "Collection " << JetTagCollection_Label[ind] <<  " ==> not found"; return;
-		}
-		//fill Inputs for All
-		for(auto& info :  *(shallowTagInfosPf)) {
```

Comment:
@andrzejnovak in the configuration given in #24522, it takes more than 300(!) events until this code is hit the first time. And that first time, the code crashes, because `shallowTagInfosPf` is not in the event.

But, there were many (hundreds?) of events before that where HLT *did* populate the `hltDeepCombinedSecondaryVertexBJetTagsInfos`, a.k.a. `shallowTagInfosPf`.

So, I have no idea what this code does/is supposed to do, but it seems to me like it is selecting very few events and exactly the wrong ones.